### PR TITLE
=htc #18539 Fix off-by-one error in response entity parsing

### LINKED ISSUE #18539: HttpResponseParser doesn't always create strict entities when easily possible
See #15686 but for responses.
(For requests this problem has already been fixed.)

### PR DESCRIPTION
**Pull request by sirthias**
Closes #18539.

/cc @jrudolph

### CODE REVIEW

**Comment by ktoso** [comment]
LGTM AFAICS, thanks!
I'll wait for @jrudolph to confirm if that's all here though.

File: akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala [hunk 1]
```diff
@@ -324,31 +335,31 @@ class ResponseParserSpec extends FreeSpec with Matchers with BeforeAndAfterAll {
...
+        .splitWhen(x ⇒ x.isInstanceOf[MessageStart] || x.isInstanceOf[EntityStreamError])
+        .via(headAndTailFlow)
+        .collect {
+          case (ResponseStart(statusCode, protocol, headers, createEntity, close), entityParts) ⇒
+            closeAfterResponseCompletion :+= close
+            Right(HttpResponse(statusCode, headers, createEntity(entityParts), protocol))
+          case (x @ (MessageStartError(_, _) | EntityStreamError(_)), _) ⇒ Left(x)
+        }
+
+    def awaitHead[T](source: Source[T, Any]): Seq[T] =
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala]: Maybe call it `collectBlocking`?

**Comment by jrudolph** [comment]
nice, good catch. Don't know why I didn't fixed that as well back then.

LGTM.

File: akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala (hunk 1 above)
Comment (sirthias) [inline akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala]: Sure, I'll do that.

**Comment by sirthias** [comment]
Updated with renaming proposed by @jrudolph.

**Comment by ktoso** [comment]
Somehow the rename did not get pushed :)
I'll merge once validation is happy, just as sanity check.

**Comment by sirthias** [comment]
Sorry, I somehow messed up the pushing yesterday.
Fixed now.

**Pull request merged**